src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,190 @@
+            break;
+        }
+        case "list": {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < taskList.getNumberOfTasks(); i++) {
+                System.out.println(i + 1 + "." + taskList.getTask(i));
+            }
+            break;
+        }
+        case "find": {
+            TaskList matchingTasks = new TaskList();
+            for (int i = 0; i < taskList.getNumberOfTasks(); i++) {
+                Task task = taskList.getTask(i);
+                String descriptionOfTask = task.description;
+                String[] splitTaskString = descriptionOfTask.split(" ");
+                for (String s : splitTaskString) {
+                    if (Objects.equals(s, after)) {
+                        matchingTasks.addTask(task);
+                    }
+                }
+            }
+            System.out.println("Here are the matching tasks in your list:");
+            for (int k = 0; k < matchingTasks.getNumberOfTasks(); k++) {
+                System.out.println(k + 1 + "." + matchingTasks.getTask(k));
+            }
+            break;
+        }
+        case "unmark": {
+            int number = Integer.parseInt(after);
+            try {
+                if (number > taskList.getNumberOfTasks()) {
+                    throw new OutOfBounds(" There is no such element :-p");
+                }
+            } catch (OutOfBounds e3) {
+                System.out.println(e3);
+                break;
+            }
+            Task toUnMarkDone = taskList.getTask(number - 1);
+            toUnMarkDone.unMarkAsDone();
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(" " + toUnMarkDone);
+            storage.updateFile(taskList);
+            break;
+        }
+        case "delete": {
+            int number = Integer.parseInt(after);
+            try {
+                if (number > taskList.getNumberOfTasks()) {
+                    throw new OutOfBounds(" There is no such element :-p");
+                }
+            } catch (OutOfBounds e3) {
+                System.out.println(e3);
+                break;
+            }
+            Task toDelete = taskList.getTask(number - 1);
+            taskList.deleteTask(number - 1);
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(" " + toDelete);
+            System.out.println("Now you have " + taskList.getNumberOfTasks() + " tasks in the list.");
+            storage.updateFile(taskList);
+            break;
+        }
+        }
+    }
+
+    /**
+     * Checks if a task has an empty description.
+     *
+     * @param checkString User input to be checked.
+     * @return True or false depending on whether description of task is empty.
+     * @throws EmptyDescription If the description of task is empty.
+     */
+    public static boolean checkEmptyDescription(String[] checkString) throws EmptyDescription {
```

Comment:
Perhaps the boolean method could be named as 'isEmpty'?
I think the method will always return true, and the usage of it is to throw an exception when the string description is empty. May be can set the method to be void so there won't be a need to return anything?